web/src/app/lists/FlatList.js

```diff
@@ -1,228 +1,230 @@
-          </ListItemSecondaryAction>
-        )}
-      </ListItem>
+      <List subheader={subheader} {...otherProps}>
+        {renderListItems()}
+
+        {/* Rendered if props.onReorder is specified */}
+        {dragPlaceholder}
+      </List>
     )
   }
 
-  renderItems() {
-    if (!this.props.items.length) {
-      return (
-        <ListItem>
-          <ListItemText
-            disableTypography
-            secondary={
-              <Typography variant='caption'>
-                {this.props.emptyMessage}
-              </Typography>
-            }
-          />
-        </ListItem>
-      )
+  /*
+   * Handles rendering either as empty message text,
+   * standard list items, list items that support
+   * drag and drop, or as a subheader list item.
+   */
+  function renderListItems() {
+    // render as empty message
+    if (!props.items.length) {
+      return <ListSubheader>{props.emptyMessage}</ListSubheader>
     }
 
-    return this.props.items.map((item, idx) => {
-      if (!this.props.onReorder) {
-        if (item.subHeader) {
-          return (
-            <ListSubheader key={idx} className={this.props.classes.background}>
-              <Typography
-                component='h2'
-                variant='subtitle1'
-                color='textSecondary'
-                data-cy='flat-list-item-subheader'
-              >
-                {item.subHeader}
-              </Typography>
-            </ListSubheader>
-          )
-        }
-        return this.renderItem(item, idx)
-      } else
+    return props.items.map((item, idx) => {
+      if (props.onReorder) {
+        // render with drag and drop
         return (
           <Draggable key={idx + item.id} draggableId={item.id} index={idx}>
             {(provided, snapshot) => {
               // light grey background while dragging non-active user
               const draggingBackground = snapshot.isDragging
-                ? this.props.classes.participantDragging
+                ? classes.dndDragging
                 : null
+
               return (
                 <div
                   ref={provided.innerRef}
                   {...provided.draggableProps}
                   {...provided.dragHandleProps}
                   className={draggingBackground}
                 >
-                  {this.renderItem(item, idx)}
+                  {renderListItem(item, idx)}
                 </div>
               )
             }}
           </Draggable>
         )
+      } else if (item.subHeader) {
+        // render list item as subheader
+        return (
+          <ListSubheader key={idx} className={classes.background}>
+            <Typography
+              component='h2'
+              variant='subtitle1'
+              color='textSecondary'
+              data-cy='flat-list-item-subheader'
+            >
+              {item.subHeader}
+            </Typography>
+          </ListSubheader>
+        )
+      } else {
```

Comment:
We should flatten these since we're exiting the function in each body. IIRC eslint has a rule to catch this specifically we should probably enable.

```js
if () return
if () return
return
```

Reply:
Sure, like I said before though: I prefer the clarity of an `if`, `if else`, and `else`, but I can make this change.